crates/prelude/src/lib.rs

```diff
@@ -40,6 +40,9 @@ extern crate alloc;
 use wasefire_applet_api as api;
 pub use wasefire_error::Error;
 
+/// Allows for main to have different return types. This should not be used publically.
```

Comment:
Let's make the whole module `doc(hidden)` since it's internal anyway. However, we can have a comment explaining why it's `pub`.
```suggestion
// This internal module is made public because it's used by exported macros called from user code.
#[doc(hidden)]
```

Reply:
Done.